doc/api/tls.md

```diff
@@ -289,7 +313,18 @@ Old clients that rely on insecure and deprecated RC4 or DES-based ciphers
 (like Internet Explorer 6) cannot complete the handshaking process with
 the default configuration. If these clients _must_ be supported, the
 [TLS recommendations] may offer a compatible cipher suite. For more details
-on the format, see the [OpenSSL cipher list format documentation].
+on the format, see the OpenSSL [cipher list format][] documentation.
+
+There are only 5 TLSv1.3 cipher suites:
+- `'TLS_AES_256_GCM_SHA384'`
+- `'TLS_CHACHA20_POLY1305_SHA256'`
+- `'TLS_AES_128_GCM_SHA256'`
+- `'TLS_AES_128_CCM_SHA256'`
+- `'TLS_AES_128_CCM_8_SHA256'`
+
+The first 3 are enabled by default. The last 2 `CCM`-based suites are supported
```

Comment:
Are we following a pattern set by someone else on this? it's an interesting exclusion. one of the main reasons CHACHA20_POLY1305_SHA256 is in there is because it might be more performant on systems that don't have hardware-enabled AES but it'd be slower on systems without it. Are we making choices here that should be left up to users & clients?

Reply:
This exactly duplicates what OpenSSL does, so I think its a safe choice. Unless I misread, these three are also what Chrome and Firefox support: https://www.ssllabs.com/ssltest/viewClient.html?name=Chrome&version=57&platform=Win%207&key=141 and  https://www.ssllabs.com/ssltest/viewClient.html?name=Firefox&version=62&platform=Win%207&key=151 .

Reply:
mm, you're right
```c
#  define TLS_DEFAULT_CIPHERSUITES "TLS_AES_256_GCM_SHA384:" \
                                   "TLS_CHACHA20_POLY1305_SHA256:" \
                                   "TLS_AES_128_GCM_SHA256"
```